java/org.travlyn.server/src/main/java/org/travlyn/server/externalapi/access/DBpediaStopRequest.java

```diff
@@ -21,23 +48,61 @@ public DBpediaStopRequest(String query) {
 
     @Override
     public Stop getResult() {
+        //check for manual name setting
+        String manualQuery = this.checkForManualName(query);
+        if (manualQuery != null){
+            query = manualQuery;
+        }
+
         Set<Pair<String, String>> params = getDefaultHeaders(query);
         params.add(new Pair<>("property", "dbo:abstract"));
         params.add(new Pair<>("property", "dbo:thumbnail"));
 
         String result = executeRequest(params);
-        JsonObject englishContent = gson.fromJson(result, JsonObject.class).getAsJsonObject("results").
-                getAsJsonArray("bindings").get(0).getAsJsonObject();
+        JsonArray resultArray;
+        JsonObject englishContent = null;
+        try {
+             resultArray = gson.fromJson(result, JsonObject.class).getAsJsonObject("results").
+                    getAsJsonArray("bindings");
+        }catch (JsonSyntaxException syntaxException){
+            //quota limit reached...exclude stop TODO
+            return null;
+        }
+
+        // filter for english language
+        for (JsonElement content : resultArray) {
+            if (content.getAsJsonObject().has("dboabstract") &&
+                    content.getAsJsonObject().getAsJsonObject("dboabstract").getAsJsonPrimitive("xml:lang").getAsString().equals("en")) {
+                englishContent = content.getAsJsonObject();
+                break;
+            }
+        }
```

Comment:
Can be outsourced to a new method and reused: `public JsonObject filterEnglish(JsonArray jsonArray) { ... }`